ocs/site_config.py

```diff
@@ -362,28 +362,36 @@ def add_arguments(parser=None):
-       site_config facility will not be used at all.""")
-    group.add_argument('--site-file', help="""Instead of the default site config, use the specified file.  Full
-       path must be specified.""")
-    group.add_argument('--site-host', help="""Override the OCS determination of what host this instance is
-       running on, and instead use the configuration for the indicated
-       host.""")
-    group.add_argument('--site-hub', help="""Override the ocs hub url (wamp_server).""")
-    group.add_argument('--site-http', help="""Override the ocs hub http url (wamp_http).""")
-    group.add_argument('--site-realm', help="""Override the ocs hub realm (wamp_realm).""")
-    group.add_argument('--instance-id', help="""Look in the SCF for Agent-instance specific configuration options,
-       and use those to launch the Agent.""")
-    group.add_argument('--address-root', help="""Override the site default address root.""")
-    group.add_argument('--registry-address', help="""Deprecated.""")
-    group.add_argument('--log-dir', help="""Set the logging directory.""")
-    group.add_argument('--working-dir', help="""Propagate the working directory.""")
+    group.add_argument('--site', help="Instead of the default site, use the "
+                       "configuration for the specified site.  The configuration is loaded "
+                       "from ``$OCS_CONFIG_DIR/{site}.yaml``.  If ``--site=none``, the "
+                       "site_config facility will not be used at all.")
+    group.add_argument('--site-file', help="Instead of the default site config, "
+                       "use the specified file. Full path must be specified.")
+    group.add_argument('--site-host', help="Override the OCS determination of "
+                       "what host this instance is running on, and instead use the "
+                       "configuration for the indicated host.")
+    group.add_argument('--site-hub', help="Override the ocs hub url (wamp_server).")
+    group.add_argument('--site-http', help="Override the ocs hub http url (wamp_http).")
+    group.add_argument('--site-realm', help="Override the ocs hub realm (wamp_realm).")
+    group.add_argument('--instance-id', help="Look in the SCF for "
+                       "Agent-instance specific configuration options, and use those to launch "
+                       "the Agent.")
+    group.add_argument('--address-root', help="Override the site default address root.")
+    group.add_argument('--registry-address', help="Deprecated.")
+    group.add_argument('--log-dir', help="Set the logging directory.")
+    group.add_argument('--working-dir', help="Propagate the working directory.")
+    group.add_argument('--crossbar-timeout', type=int, default=10, help="Length of time in seconds "
+                       "that the Agent will try to reconnect to the crossbar server before "
+                       "shutting down. Note this is set per Agent in an instance's arguments list.")
```

Comment:
I think it's important that the default can be set as a global parameter in the SCF.  Similar to --site-*, the command line arg would override the SCF default.

Reply:
Alright, I can do that once I figure out how to add a global argument to the site config system. Do you imagine this at the host level, or higher?

Reply:
I figured this out, it'll still default to 10 seconds, but can be set at the host level, as an agent argument, or on the commandline, with precedence higher as you get closer to the Agent/CLI.

Done in https://github.com/simonsobs/ocs/pull/337/commits/1ff8ded07fe1ea48391d84a351db3761568aa293 and documented in https://github.com/simonsobs/ocs/pull/337/commits/91239f0aaaa25e308e2ab68f56a302c73d9544e6.